lib/web_browser.pm

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: common functions for FIPS web browser tests
+# Maintainer: llzhao <llzhao@suse.com>
+
+package web_browser;
+
+use base Exporter;
+use Exporter;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+use registration qw(add_suseconnect_product register_product);
+
+our @EXPORT = qw(
+  setup_web_browser_env
+  run_web_browser_text_based
+);
+
+# Setup test envs: register PackageHub and check FIPS pattern installed
+sub setup_web_browser_env {
+    my $ret = 0;
+
+    # Check the register status for reference
+    script_run("SUSEConnect --list-extension | grep 'SUSE Package Hub' | grep '(Activated)'");
+    $ret = script_run("output=`SUSEConnect -s | grep 'PackageHub'`; echo \${output##*PackageHub} | cut -d ':' -f4 | grep 'Registered'");
+    if ($ret ne 0) {
+        # Workaround: in some cases extensions may be in false "Activated" status,
+        # then just re-register it and do not check the return value
+        if (is_sle) {
+            register_product();
+            my $version = get_required_var('VERSION') =~ s/([0-9]+)-SP([0-9]+)/$1.$2/r;
+            my $arch    = get_required_var('ARCH');
+            script_run("SUSEConnect -d -p PackageHub/$version/$arch", 300);
+            script_run("SUSEConnect -p PackageHub/$version/$arch",    300);
+        }
+    }
+    zypper_call("--no-refresh --no-gpg-checks search -it pattern fips");
+}
+
+# Run text based web browser with options
+# $browser: The name of text based web browser: w3m/links/lynx
+# $options: command line options
+sub run_web_browser_text_based {
+    my ($browser, $options) = @_;
+
+    my %https_url = (
+        google => "https://www.google.com/ncr",
+        suse   => "https://www.suse.com/",
+        OBS    => "https://build.opensuse.org/",
```

Comment:
I do not recommend to use these website for testing purpose, because they are mutable. This suggestion is not mandatory but maybe we should consider.

Reply:
Agree with your concerns.
But I will keep the urls as I moved them from the "w3m_https.pm" and also I have no better ones.

Reply:
at least I would remove google , in my previous company such automation tests which using google as test page cause that whole office was not able to do regular googling because Google just blocking whole company due to suspicious behavior ...

Reply:
ah might be irrelevant , haven't notice that you just loading page once so low chances that Google will treat it anyhow bad

Reply:
Thank you so much, good to know your idea.
BTW, if we want to change google url to other url we have to add new needles or delete unneeded google needles, we have to open a new poo to fix it.
So, I will keep the urls unchanged this time.